sklearn/utils/_param_validation.py

```diff
@@ -364,7 +364,7 @@ class Interval(_Constraint):
 
     Parameters
     ----------
-    type : {numbers.Integral, numbers.Real}
+    type : {numbers.Integral, numbers.Real, "real_not_int"}
         The set of numbers in which to set the interval.
```

Comment:
Would it be good to have a description of these internally in the docstring? Just for community-devs that aren't familiar w/ what each of these are intended to mean?

Reply:
I added a description of the new option.

A short description of all the constraints and what they represent can be found here https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/_param_validation.py#L28

Reply:
Looks good! 

I was just mentioning mainly the `real_not_int` option, not the others.